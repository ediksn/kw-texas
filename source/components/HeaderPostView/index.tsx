import React, { ReactNode } from 'react'
import { TouchableHighlight, View, ViewStyle } from 'react-native'
import { styles } from './styles'

interface Props {
  children: ReactNode
  statusBarHeight?: number
  leftButton?: ReactNode
  onClickLeft?: () => void
  style?: ViewStyle
}

interface LeftButtonProps {
  leftButton?: ReactNode
  onClickLeft?: () => void
}

const LeftButton = ({ leftButton, onClickLeft }: LeftButtonProps) => (
  <>
    {leftButton && (
      <TouchableHighlight
        style={styles.touchableHighlight}
        activeOpacity={0.6}
        underlayColor='transparent'
        onPress={onClickLeft}
      >
        {leftButton}
      </TouchableHighlight>
    )}
  </>
)

const HeaderPostView = ({ children, statusBarHeight, leftButton, onClickLeft, style }: Props) => {
  return (
    <View style={[styles.container, style, { paddingTop: statusBarHeight }]}>
      <LeftButton leftButton={leftButton} onClickLeft={onClickLeft} />
      {children}
    </View>
  )
}

export default HeaderPostView
