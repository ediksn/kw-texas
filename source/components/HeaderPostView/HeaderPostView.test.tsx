import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import HeaderPostView from '~/components/HeaderPostView/index'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

describe('Render Header Post View Test', () => {
  let component: RenderAPI

  beforeEach(() => {
    component = render(
      <HeaderPostView>
        <></>
      </HeaderPostView>
    )
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
