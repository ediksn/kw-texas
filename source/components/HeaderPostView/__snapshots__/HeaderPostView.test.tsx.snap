// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Post View Test Match with the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "elevation": 8,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "maxWidth": "100%",
        "paddingVertical": "2%",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
        "width": "100%",
      },
      undefined,
      Object {
        "paddingTop": undefined,
      },
    ]
  }
/>
`;
